components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIDefinitionFromSwagger20.java

```diff
@@ -72,12 +76,89 @@
 public class APIDefinitionFromSwagger20 implements APIDefinition {
 
     private static final Logger log = LoggerFactory.getLogger(APIDefinitionFromSwagger20.class);
+    private static Map<String, Map<String, String>> localConfigMap = new HashMap<>();
```

Comment:
because for each request it refers to a new instance of APIDefinitionFromSwagger20